src/Chapter2.hs

```diff
@@ -842,7 +874,11 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate n xs
+  | null xs   = []
+  | n < 0     = []
+  | otherwise = drop n (take (n+length xs) (cycle xs))
```

Comment:
Nice one! 👏🏼 

You can also use a trick for efficiency: you can notice that when `n` is big, this function will do much more cycles. So you can use `mod` on the length of the list 👍🏼